packages/ag-charts-community/src/chart/axis/categoryAxis.ts

```diff
@@ -35,12 +35,17 @@ export class CategoryAxis extends CartesianAxis<BandScale<string | object>> {
     }
 
     normaliseDataDomain(d: (string | object)[]): (string | object)[] {
-        // Prevent duplicate categories.
-        const valuesSet: Record<string, (typeof d)[number]> = {};
-        for (const next of d) {
-            valuesSet[String(next)] ??= next;
+        const domain = [];
+        const uniqueValues = new Set();
+        for (const v of d) {
+            const key = v instanceof Date ? v.getTime() : v;
```

Comment:
Will it fail for `{ toString: () ='abc' }`?

Reply:
Nevermind 🤦

Reply:
no

Reply:
https://github.com/ag-grid/ag-charts/pull/67#discussion_r1280671452